components/eventing-controller/controllers/subscription/jetstream/reconciler.go

```diff
@@ -145,10 +146,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 
 	// Synchronize Kyma subscription to JetStream backend
 	if err := r.Backend.SyncSubscription(desiredSubscription); err != nil {
+		result := ctrl.Result{}
 		if syncErr := r.syncSubscriptionStatus(ctx, desiredSubscription, statusChanged, err); syncErr != nil {
-			return ctrl.Result{}, syncErr
+			return result, syncErr
 		}
-		return ctrl.Result{}, err
+		// Requeue the Request to reconcile it again if there are no NATS Subscriptions synced
+		if missingSubscriptionErr(err) {
+			result = ctrl.Result{RequeueAfter: jetstream.RequeueDuration}
+			err = nil
```

Comment:
why are we setting err to nil?

Reply:
because otherwise it will output an error in the controller. Omitting this error was the main point of the PR. But the error will still be propagated to the Subscription and it will be set to unready.